functions/Add-DbaPushSubscription.ps1

```diff
@@ -0,0 +1,94 @@
+function Add-DbaPushSubscription {
+    <#
+    .SYNOPSIS
+        Adds a new subscriber to an existing publication
+
+    .DESCRIPTION
+        Creates a new replication subscription to the existing specified publication
+
+    .PARAMETER publisherName
+        The target SQL Server(s) which is the publisher.
+
+    .PARAMETER subscriberName
+        The target SQL Server(s) which is the subscriber.
+
+    .PARAMETER publicationName
+        The name of the publication on the publisher.
+
+    .PARAMETER pubDatabaseName
+        The name of the publications database on the publisher.
+
+    .PARAMETER subDatabaseName
+        The name of the database on the subscriber.
+
+    .PARAMETER subSyncType
+        The subscription sync type. e.g. backup, snapshot, none
+
+
+    .NOTES
+        Tags: Replication
+        Author: Gareth Newman
+
+        Website: https://dbatools.io
+        Copyright: (c) 2018 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+    .LINK
+        https://dbatools.io/Add-DbaPushSubscription
+
+    .EXAMPLE
+        PS C:\> Add-DbaPushSubscription -publisherName "server1\dev2016" -subscriberName "server2\dev2016" -publicationName "avworkspub1" -pubDatabaseName "AdventureWorks2012" -subDatabaseName "replicatedAdventureWorks2012" -subSyncType "none"
+
+        Creates a new subscription to "avworkspub1" on server1\dev2016 to server2\dev2016 with a subscription sync type of none.
+        The published database is AdventureWorks2012 on server1\dev2016, the subscriber database is replicatedAdventureWorks2012 on server2\dev2016.
+
+    #>
+    Param(
+        [Parameter(Mandatory)]
+        $publisherName,
+        $subscriberName,
+        $publicationName,
+        $pubDatabaseName,
+        $subDatabaseName,
+        $subSyncType
+    )
+
+    # create the server connection object
+    try {
+        $srv = New-Object Microsoft.SqlServer.Management.Smo.Server $publisherName
```

Comment:
Thanks so much for the command! Please use the templates at [dbatools.io/templates](https://dbatools.io/templates). In particular, this one: https://github.com/sqlcollaborative/dbatools-templates/blob/master/Get-DbaSmoObject.ps1

Please add Rep this - `Add-DbaRepTherest....` but what do you think about `Add-DbaRepSubscription -Type Push` so you can support all the diff subscription types from one command? (asking, not commanding, im don't know enough about replication to say, really)